src/plugins/htmlVideoPlayer/plugin.js

```diff
@@ -706,124 +1070,125 @@ define(['browser', 'require', 'events', 'apphost', 'loading', 'dom', 'playbackMa
-
-                    if (currentTrack.label.indexOf('manualTrack') !== -1) {
-                        currentTrack.mode = 'disabled';
+                const allTracks = videoElement.textTracks || []; // get list of tracks
+                for (const track of allTracks) {
+                    if (track.label.includes('manualTrack')) {
+                        track.mode = 'disabled';
                     }
                 }
             }
 
-            customTrackIndex = -1;
-            currentClock = null;
-            self._currentAspectRatio = null;
+            this.#customTrackIndex = -1;
+            this.#currentClock = null;
+            this._currentAspectRatio = null;
 
-            var octopus = currentSubtitlesOctopus;
+            const octopus = this.#currentSubtitlesOctopus;
             if (octopus) {
                 octopus.dispose();
             }
-            currentSubtitlesOctopus = null;
+            this.#currentSubtitlesOctopus = null;
 
-            var renderer = currentAssRenderer;
+            const renderer = this.#currentAssRenderer;
             if (renderer) {
                 renderer.setEnabled(false);
             }
-            currentAssRenderer = null;
+            this.#currentAssRenderer = null;
         }
 
-        self.destroyCustomTrack = destroyCustomTrack;
-
-        function fetchSubtitlesUwp(track, item) {
+        /**
+         * @private
+         */
+        fetchSubtitlesUwp(track) {
             return Windows.Storage.StorageFile.getFileFromPathAsync(track.Path).then(function (storageFile) {
-                return Windows.Storage.FileIO.readTextAsync(storageFile).then(function (text) {
-                    return JSON.parse(text);
-                });
+                return Windows.Storage.FileIO.readTextAsync(storageFile);
+            }).then(function (text) {
+                return JSON.parse(text);
             });
         }
 
-        function fetchSubtitles(track, item) {
+        /**
+         * @private
+         */
+        async fetchSubtitles(track, item) {
             if (window.Windows && itemHelper.isLocalItem(item)) {
-                return fetchSubtitlesUwp(track, item);
+                return this.fetchSubtitlesUwp(track, item);
             }
 
-            incrementFetchQueue();
-            return new Promise(function (resolve, reject) {
-                var xhr = new XMLHttpRequest();
-
-                var url = getTextTrackUrl(track, item, '.js');
-
-                xhr.open('GET', url, true);
-
-                xhr.onload = function (e) {
-                    resolve(JSON.parse(this.response));
-                    decrementFetchQueue();
-                };
-
-                xhr.onerror = function (e) {
-                    reject(e);
-                    decrementFetchQueue();
-                };
+            this.incrementFetchQueue();
+            try {
+                const response = await fetch(getTextTrackUrl(track, item, '.js'));
```

Comment:
@thornbill said that errors from `fetch` should be handled https://github.com/jellyfin/jellyfin-web/pull/1503#discussion_r460453048

Maybe something like this?
```suggestion
                const response = await fetch(getTextTrackUrl(track, item, '.js'));

                if (!response.ok) {
                    throw new Error(response.status);
                }
```

Reply:
I'll throw the whole response
```suggestion
                const response = await fetch(getTextTrackUrl(track, item, '.js'));
                
                if (!response.ok) {
                    throw new Error(response);
                }
```